packages/stokhos/src/sacado/kokkos/vector/tpetra/Stokhos_Tpetra_ETI_Helpers_MP_Vector_ENSEMBLE_SIZE_DEVICE.hpp

```diff
@@ -46,23 +46,43 @@
 #define INSTANTIATE_MP_VECTOR_STORAGE(INSTMACRO, STORAGE, LO, GO, N)      \
   INSTMACRO( Sacado::MP::Vector<STORAGE>, LO, GO, N )
 
+#define INSTANTIATE_MP_VECTOR_STORAGE_SD(INSTMACRO, STORAGE, N)      \
+  INSTMACRO( Sacado::MP::Vector<STORAGE>, N )
+
 #define INSTANTIATE_MP_VECTOR_SFS_SLND(INSTMACRO, S, L, NUM, D, LO, GO, N) \
   typedef Stokhos::StaticFixedStorage<L,S,NUM,D::execution_space> SFS_ ## L ## _ ## S ## _ ## NUM ## _ ## D; \
   INSTANTIATE_MP_VECTOR_STORAGE(INSTMACRO, SFS_ ## L ## _ ## S ## _ ## NUM ## _ ## D, LO, GO, N)
 
+#define INSTANTIATE_MP_VECTOR_SFS_SLND_SD(INSTMACRO, S, L, NUM, D, N) \
+  typedef Stokhos::StaticFixedStorage<L,S,NUM,D::execution_space> SFS_ ## L ## _ ## S ## _ ## NUM ## _ ## D ## _2; \
+  INSTANTIATE_MP_VECTOR_STORAGE_SD(INSTMACRO, SFS_ ## L ## _ ## S ## _ ## NUM ## _ ## D ## _2, N)
+
 #define INSTANTIATE_MP_VECTOR_SFS_SLD(INSTMACRO, S, L, D, LO, GO, N) \
   INSTANTIATE_MP_VECTOR_SFS_SLND(INSTMACRO, S, L,  @ENSEMBLE_SIZE@, D, LO, GO, N)
 
+#define INSTANTIATE_MP_VECTOR_SFS_SLD_SD(INSTMACRO, S, L, D, N) \
+  INSTANTIATE_MP_VECTOR_SFS_SLND_SD(INSTMACRO, S, L,  @ENSEMBLE_SIZE@, D, N)
+
 #define INSTANTIATE_MP_VECTOR_S_D(INSTMACRO, D, LO, GO, N) \
   INSTANTIATE_MP_VECTOR_SFS_SLD(INSTMACRO, double, int, D, LO, GO, N)
 
+#define INSTANTIATE_MP_VECTOR_S_D_SD(INSTMACRO, D, N) \
+  INSTANTIATE_MP_VECTOR_SFS_SLD_SD(INSTMACRO, double, int, D, N)
+
 #define INSTANTIATE_MP_VECTOR_S(INSTMACRO, LO, GO, N) \
   typedef Stokhos::DeviceForNode<N>::type DFN_CPU_ ## LO ## _ ## GO ## _ ## N; \
   INSTANTIATE_MP_VECTOR_S_D(INSTMACRO, DFN_CPU_ ## LO ## _ ## GO ## _ ## N, LO, GO, N)
 
+#define INSTANTIATE_MP_VECTOR_S_SD(INSTMACRO, N) \
+  typedef Stokhos::DeviceForNode<N>::type DFN_CPU_ ## N; \
```

Comment:
I don't think we actually need `DeviceForNode` any more, since all the Node types remaining are post-Kokkos-refactor types that have `device_type`, `execution_space`, and `memory_space` typedefs.  This is not relevant to the current PR, though.